src/main/haskell/kore/src/Kore/Substitution/Class.hs

```diff
@@ -47,12 +54,12 @@ class MapClass s v t => SubstitutionClass s v t where
     substitutionTermsFreeVars :: s v t -> Set.Set v
 
 {-'SubstitutionAndQuantifiedVars' is a substitution which can hold more free
-variables than its terms can.  'quantifiedVars' is used to track the free
+variables than its terms can.  'freeVars' is used to track the free
 variables in a substitution context.
 -}
 data SubstitutionAndQuantifiedVars s var pat = SubstitutionAndQuantifiedVars
     { substitution   :: s var pat
-    , quantifiedVars :: Set.Set var
+    , freeVars :: Set.Set var
```

Comment:
How well does that work for a small number of variables?
I wasn't trying to make that sort of change, I just renamed the field while trying to understand the code enough to remove the old traversal scheme.

Reply:
Both `Set` and `HashSet` are going to have lookup that is `O(k)` in the key length `k`, if that's what you mean. I only bring it up because I noticed it now and I know it's what some other interpreters do; I don't mean to say it needs to be addressed as part of this PR.

Reply:
Wait,  `HashSet` can't be truly `O(1)` without mutability, right?

Reply:
That's true; the documentation notes

> Many operations have a average-case complexity of O(log n). The implementation uses a large base (i.e. 16) so in practice these operations are constant time.